flight/Modules/VtolPathFollower/vtol_follower_fsm.c

```diff
@@ -342,10 +358,21 @@ static int32_t vtol_fsm_static()
 			return ret_val;
 	}
 
-	current_count++;
-
-	if ((timer_duration > 0) && ((current_count - set_time_count) * DT) > timer_duration) {
-		vtol_fsm_inject_event(FSM_EVENT_TIMEOUT);
+	/* If there's a timer... */
```

Comment:
This logic should be moved into a function that just checks the timer for expiration. Will also help for refactoring into library later, right?